test/auto_parallel/semi_auto_parallel_dist_to_static_mlp.py

```diff
@@ -0,0 +1,221 @@
+            for batch_id, (image, label) in enumerate(data_loader()):
+                out = layer(image)
+                loss = loss_fn(out, label)
+                loss_list.append(loss.numpy())
+                loss.backward()
+
+                opt.step()
+                opt.clear_grad()
+        return np.array(loss_list)
+
+    def test_dp_demo_net(self):
+        mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
+        self.set_random_seed(self._seed)
+        data_loader = self.create_data_loader()
+
+        self.set_random_seed(self._seed)
+        dy_layer = DemoNet(mesh, "dy_dp_demonet", shard_input=True)
+        dy_opt = paddle.optimizer.SGD(
+            learning_rate=0.1, parameters=dy_layer.parameters()
+        )
+
+        self.set_random_seed(self._seed)
+        dy2static_layer = DemoNet(
+            mesh, "dy2static_dp_demonet", shard_input=True
+        )
+        dy2static_opt = paddle.optimizer.SGD(
+            learning_rate=0.1, parameters=dy2static_layer.parameters()
+        )
+
+        dy_losses = self.run_dynamic(dy_layer, dy_opt, data_loader)
+        dy2static_losses = self.run_dy2static(
+            dy2static_layer, dy2static_opt, data_loader
+        )
+
+        # Check the loss values. Different from dygraph mode, when
+        # the model is trained in dy2static mode, the loss values
+        # are not the average of the losses of all processes, so
+        # we should get the average loss first.
+        paddle.disable_static()
+        pd_partial_loss = paddle.to_tensor(dy2static_losses)
+        pd_loss_list = []
+        dist.all_gather(pd_loss_list, pd_partial_loss)
+        np_dy2static_loss_list = [loss.numpy() for loss in pd_loss_list]
+        np_dy2static_loss = np.array(np_dy2static_loss_list)
+        np_dy2static_loss = np.mean(np_dy2static_loss, axis=0)
+
+        np.testing.assert_allclose(dy_losses, np_dy2static_loss, rtol=1e-6)
+
+    def test_mp_demo_net(self):
+        mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
+        self.set_random_seed(self._seed)
+        data_loader = self.create_data_loader()
+
+        self.set_random_seed(self._seed)
+        dy_layer = DemoNet(mesh, "dy_mp_demonet", shard_weight=True)
+        dy_opt = paddle.optimizer.SGD(
+            learning_rate=0.1, parameters=dy_layer.parameters()
+        )
+
+        self.set_random_seed(self._seed)
+        dy2static_layer = DemoNet(
+            mesh, "dy2static_mp_demonet", shard_weight=True
+        )
+        dy2static_opt = paddle.optimizer.SGD(
+            learning_rate=0.1, parameters=dy2static_layer.parameters()
+        )
+
+        dy_losses = self.run_dynamic(dy_layer, dy_opt, data_loader)
+        dy2static_losses = self.run_dy2static(
+            dy2static_layer, dy2static_opt, data_loader
+        )
+
+        np.testing.assert_allclose(dy_losses, dy2static_losses, rtol=1e-6)
+
+    # python -m paddle.distributed.launch --devices=0,1 semi_auto_parallel_static_decorate_mlp.py
```

Comment:
`semi_auto_parallel_static_decorate_mlp.py` should be `semi_auto_parallel_dist_to_static_mlp.py`?